### PR TITLE
Fix issues with dynamic key reload for imported/derived keymanager

### LINKED ISSUE #8577: Validator client generates errors if it had validators but they all exit
# 🐞 Bug Report

### Description

If a client was processing active validators but they all exit, once it has no more active, it continues to generate the following error:

`time="2021-03-08 19:15:35" level=error msg="Failed to update assignments" error="rpc error: code = InvalidArgument desc = no attester slots provided" prefix=validator`

### Has this worked before in a previous version?

Unknown but not likely..

## 🔬 Minimal Reproduction

Spin up a client with a single active validator; exit it; wait for it to exit.

## 🔥 Error

<pre><code>

time="2021-03-08 19:15:35" level=error msg="Failed to update assignments" error="rpc error: code = InvalidArgument desc = no attester slots provided" prefix=validator

</code></pre>

## 🌍  Your Environment

**Operating System:**

Ubuntu 18.04

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
prysm v1.2.1
  </code>
</pre>

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

Makes all scenarios outlined in https://docs.google.com/document/d/1fLdUi6AHQ5-wYOy6DCzW5JlLlq3VuI275-KoyEZ2Pmw/edit#heading=h.m87ybthlvflh work correctly.

**High-level overview:**

I added a new case in the runner, where I listen to account changes via a channel. After such changes occur, I handle key reload by logging out statuses and returning if any keys are active. If there are no active keys after the reload, I call `WaitForActivation` passing the channel as a way of reacting to reload inside the function. Note that the call to `WaitForActivation`, which happens before the main loop, does not need an external channel, so it manages its own when `nil` is passed in.

I defined a `validatorStatus` struct abstraction to be able to log out statuses from both `WaitForValidation` and `HandleKeyReload`.

The PR also fixes a bug in `validator/keymanager/imported/refresh.go`. Currently the event publishes old keys instead of new ones.

**Which issues(s) does this PR fix?**

Part of #8572
Fixes #8577

**Other notes for review**

N/A

### CODE REVIEW

**Comment by rkapka** [comment]
I would like a review from @prestonvanloon and @rauljordan before merging this.

File: validator/client/runner.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func run(ctx context.Context, v iface.Validator) {
 		if err != nil {
 			log.Fatalf("Could not determine if beacon node synced: %v", err)
 		}
-		err = v.WaitForActivation(ctx)
+		err = v.WaitForActivation(ctx, nil)
```
Comment (rauljordan) [inline validator/client/runner.go:78]: This is a little weird without context, I would recommend a comment like
`nil /* something something */`

File: validator/client/wait_for_activation.go [hunk 2]
```diff
@@ -13,22 +12,25 @@ import (
...
-	defer func() {
-		sub.Unsubscribe()
-		close(accountsChangedChan)
-	}()
+	if accountsChangedChan == nil {
+		accountsChangedChan = make(chan [][48]byte)
+		sub := v.GetKeymanager().SubscribeAccountChanges(accountsChangedChan)
+		defer func() {
+			sub.Unsubscribe()
+			close(accountsChangedChan)
```
Comment (rauljordan) [inline validator/client/wait_for_activation.go:31]: This defer will close the channel by the time it makes it into v.waitForActivation. Is this intended behavior?

Comment (rkapka) [reply validator/client/wait_for_activation.go:31]: The function will not return until `waitForActivation` completes, so the channel will not get closed. This PR only adds an `if` statement, so it wouldn't have worked in the first place.

Comment (rauljordan) [reply validator/client/wait_for_activation.go:31]: You are totally right

**Review by rauljordan — APPROVED** [review]

**Pull request merged**